lstests/core/logger.py

```diff
@@ -0,0 +1,345 @@
+#! /usr/bin/env python3
+
+# standard libraries
+import abc
+import dataclasses
+import inspect
+import os.path
+import queue
+import threading
+import time
+import typing
+# user-defined modules
+if __package__:
+    from . import color
+    from . import helper
+else:
+    import color
+    import helper
+
+try:
+    VERTICAL_BAR = "│"
+except UnicodeEncodeError:
+    VERTICAL_BAR = "|"
+
+
+class LogLevel(int):
```

Comment:
Thank you for the comment. Initially when I wrote the code, I intended to inherit from `Enum.IntEnum`. But later I found it's too restrictive to serve my purpose. I wanted to achieve both ends: (1) have fixed, named levels, so that, say, `DEBUG` is equivalent to `20` under the hood, (2) provide flexibility by allowing users to supply, say, `38` as the log level -- and this `38` should be comparable (for greater than or less than) with `DEBUG`. So eventually I chose to inherit from `int`. I guess `Enum.IntFlag` has the same restriction as does `Enum.IntEnum`.